staging/src/k8s.io/apiserver/pkg/server/filters/timeout.go

```diff
@@ -119,6 +120,23 @@ func (t *timeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		}
 		return
 	case <-after:
+		defer func() {
```

Comment:
How about using `resultCh := make(chan interface{}, 1)`, maybe there is no need to create a new goroutine

Reply:
That can swallow a non-nil panic in time out cases, which we don't want

Reply:
Maybe we can call `utilruntime.HandleError`  just before `resultCh <- err`